src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1143,6 +1053,29 @@ static void setResponseMetadata(ContextAttribute* caReq, ContextAttribute* caRes
 
 /* ****************************************************************************
 *
+* howManyAttrs -
+*
+* Returns the number of occurences of the attrName attribute (e.g. "A1") in the attrs
+* BSON (taken from DB). This use to be 1, except in the case of using ID.
+*
+*/
+static unsigned int howManyAttrs(BSONObj& attrs, std::string& attrName)
+{
+  unsigned int c = 0;
+  std::set<std::string> attrNames;
+  attrs.getFieldNames(attrNames);
+  for(std::set<std::string>::iterator i = attrNames.begin(); i != attrNames.end(); ++i)
+  {
+    if(basePart(*i) == attrName)
```

Comment:
'if' is not a function but a builtin.
Builtins, like 'while', 'for' and 'if' should have a space between the keyword and the parenthesis:

    if (basePart(*i) == attrName) ...